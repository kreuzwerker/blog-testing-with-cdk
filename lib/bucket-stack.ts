import * as cdk from '@aws-cdk/core';
import * as s3 from '@aws-cdk/aws-s3';

export class BucketStack extends cdk.Stack {
  constructor(environment: string, scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const bucket = new s3.Bucket(this, 'Bucket', {
      bucketName: `my-bucket-${environment}`,
      removalPolicy: cdk.RemovalPolicy.DESTROY
    });
    cdk.Tags.of(bucket).add('abc', '123')
    cdk.Tags.of(bucket).add('environment', environment)
    //cdk.Tags.of(bucket).add('xyz', '123')

    new s3.Bucket(this, 'OtherBucket');
  }
}
