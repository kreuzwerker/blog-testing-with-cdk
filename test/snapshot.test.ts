import { SynthUtils } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
 
import { BucketStack } from '../lib/bucket-stack';
 
test('bucket stack', () => {
 const app = new cdk.App();
 const stack = new BucketStack('foo', app, 'TestBucketStack');
 expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
