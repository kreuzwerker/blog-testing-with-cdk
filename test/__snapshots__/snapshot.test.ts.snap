// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bucket stack 1`] = `
Object {
  "Resources": Object {
    "Bucket83908E77": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketName": "my-bucket-foo",
        "Tags": Array [
          Object {
            "Key": "abc",
            "Value": "123",
          },
          Object {
            "Key": "environment",
            "Value": "foo",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
